### PR TITLE
AMBARI-24743. Cleanup logsearch: add/cleanup javadoc.

### PR DESCRIPTION
**Pull request by oleewere**
# What changes were proposed in this pull request?
- add more javadocs, especially for logfeeder plugin api
- cleanup and fix javadoc warnings and errors
- add new make build task to generate javadocs
- a small refactoring to simplify some code

## How was this patch tested?
UTs passed.
javadoc generation was used with "make javadoc"

please review @zeroflag @swagle @adoroszlai @g-boros @kasakrisz

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/4/
Test PASSed.

File: ambari-logsearch-config-api/src/main/java/org/apache/ambari/logsearch/config/api/LogLevelFilterManager.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
    *
    * @param clusterName The name of the cluster where the logs are.
    * @param filters The log level filters to set.
-   * @throws Exception
+   * @throws Exception error during setting the log lelev filter
```
Comment (adoroszlai) [inline ambari-logsearch-config-api/src/main/java/org/apache/ambari/logsearch/config/api/LogLevelFilterManager.java]: typo: `lelev`

**Comment by oleewere** [comment]
@adoroszlai right, but if I'm providing any javadoc to a method i need to add descriptions to the params in order to avoid warnings - although the names mostly meaningful.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/5/
Test PASSed.

File: ambari-logsearch-server/src/main/java/org/apache/ambari/logsearch/common/ExternalServerClient.java [hunk 2]
```diff
@@ -50,17 +50,18 @@
 
   /**
    * Send GET request to an external server
+   * @param loginUrl external url
+   * @param klass response object type
```
Comment (g-boros) [inline ambari-logsearch-server/src/main/java/org/apache/ambari/logsearch/common/ExternalServerClient.java]: typo: klass?

File: ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/output/OutputSolr.java [hunk 3]
```diff
@@ -56,6 +56,34 @@
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 
+/**
+ * Ship (transformed) input data to solr destination. Works with both solr cloud mode or providing static solr url(s).
+ * In Solr cloud mode Log Feeder will manage and listen a ZooKeeper connection. If tthere are too many Log Feeder nodes that can mean
```
Comment (g-boros) [inline ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/output/OutputSolr.java]: typo: tthere

File: ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/input/monitor/DockerLogFileUpdateMonitor.java [hunk 4]
```diff
@@ -31,14 +31,15 @@
...
- * - input has monitored already - container stopped - log is not too old -> keep in the monitored list <br/>
- * - input has monitored already - container does not exist - remove it from the monitoed list (and all other input with the same log type) <br/>
+ * <pre>
+ * Use cases:
+ * - input has not monitored yet - found new container: start monitoring it
+ * - input has not monitored yet - found new stopped container: start monitoring it
+ * - input has not monitored yet - found new stopped container but log is too old: do not monitoring it
+ * - input has monitored already - container stopped - if it's stopped for too long time : remove it from the monitoed list
+ * - input has monitored already - container stopped - log is not too old: keep in the monitored list
+ * - input has monitored already - container does not exist - remove it from the monitoed list (and all other input with the same log type)
```
Comment (g-boros) [inline ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/input/monitor/DockerLogFileUpdateMonitor.java]: typo: monitoed

**Pull request merged**